crf/src/test/java/org/sagebase/crf/step/active/HeartRateSampleProcessorTest.kt

```diff
@@ -301,11 +262,13 @@ class HeartRateSampleProcessorTest {
         // Only check some of the results b/c the calculations are for a resting heart rate and all use
         // initial estimate or else they are invalid (Initial windows).
         for (ii in 0 until Math.min(10, expectedOutputs.size)) {
-            val (hr, confidence) = processor.calculateHRFromFilteredSamples(inputChunks[ii], samplingRate)
+            //let filteredData = processor.getFilteredSignal(inputChunks[ii], samplingRate: roundedSamplingRate)
+            val filteredData = processor.getFilteredSignal(inputChunks[ii].toList(), roundedRate)
+            val (hr, confidence) = processor.calculateHRFromFilteredSamples(filteredData, samplingRate)
             val expectedHR = expectedOutputs[ii].hr
             val expectedConfidence = expectedOutputs[ii].confidence
-            assertEquals(expectedHR, hr,0.000001)
-            assertEquals(expectedConfidence, confidence, 0.000001)
+            assertEquals(expectedHR, hr,0.0001)
```

Comment:
any idea why the values are matching less precisely than before? (I'm assuming that's why you changed this)

Reply:
It's because there is an error in the R-implementation that has not been addressed and which I didn't want to propagate into the mobile implementation.